src/status_im/contexts/wallet/swap/setup_swap/view.cljs

```diff
@@ -265,7 +277,22 @@
                                                         (fn [input-state]
                                                           (controlled-input/set-input-value
                                                            input-state
-                                                           max-value)))))]
+                                                           max-value)))))
+        on-refresh-swap-proposal                    (fn []
+                                                      (when (and valid-pay-input?
+                                                                 (not loading-swap-proposal?))
+                                                        (fetch-swap-proposal
+                                                         {:amount       pay-input-amount
+                                                          :valid-input? valid-pay-input?})))]
+    (rn/use-effect (fn []
+                     (when refetch-timeout
+                       (js/clearInterval refetch-timeout)
+                       (set-refetch-timeout nil))
+                     (when (or swap-proposal error-response)
+                       (set-refetch-timeout (js/setInterval
+                                             on-refresh-swap-proposal
+                                             constants/swap-proposal-refresh-interval-ms))))
+                   [swap-proposal error-response])
```

Comment:
We might need to clear the interval on unmount. WDYT?

Reply:
On navigate back we dispatch `:wallet/clean-swap-proposal` before navigating which removed `swap-proposal` and `error-response` keys from db, that would trigger this effect and the interval should be cleared, so _I think_ it is not needed. But I can add it on unmount too.

Reply:
Added the clearInterval on unmount too!

Reply:
Thanks, Brian! I was concerned about the swipe left-to-right gesture to close/return to the previous screen where the user will skip the `on-close` method call.